src/App.tsx

```diff
@@ -1,18 +1,31 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
+import { ItemsOnPage } from './types/itemsOnPage';
+import { items } from './constants/items';
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage]
+    = useState<ItemsOnPage>(ItemsOnPage.Five);
+  const [currentPage, setCurrentPage] = useState(1);
+  const firstItemIndexOnPage = (currentPage * itemsPerPage) - itemsPerPage;
+  const lastItemIndexOnPage = (currentPage * itemsPerPage);
+
+  const handleSelectorChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    setItemsPerPage(+e.target.value);
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${firstItemIndexOnPage + 1} - ${
+          lastItemIndexOnPage > 42
+            ? 42
```

Comment:
```suggestion
            ? items.length
```

it is better to not hardcode total items amount